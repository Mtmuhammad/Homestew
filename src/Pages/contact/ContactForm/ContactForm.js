import React from "react";
import "./ContactForm.scss";
import contact from "../../../Assets/Images/home/Contact.jpg";

const ContactForm = () => {
  return (
    <div className="container">
      <div className="row justify-content-center">
        <div className="col-md-12">
          <div className="wrapper">
            <div className="row no-gutters">
              <div className="col-lg-6">
                <div className="contact-wrap w-100 p-md-5 p-4">
                  <h3 className="mb-3">Contact Us</h3>
                  <p className="mb-4">
                    We're open for any suggestion or just to have a chat
                  </p>
                  <div className="row mb-4">
                    <div className="col-md-4">
                      <div className="w-100 d-flex align-items-start">
                        <div className="text">
                          <p>
                            <span>Address:</span> 198 West 21th Street, Suite
                            721 New York NY 10016
                          </p>
                        </div>
                      </div>
                    </div>
                    <div className="col-md-4">
                      <div className="w-100 d-flex align-items-start">
                        <div className="text">
                          <p>
                            <span>Email:</span>{" "}
                            <a href="mailto:info@yoursite.com">
                              info@homestew.com
                            </a>
                          </p>
                        </div>
                      </div>
                    </div>
                    <div className="col-md-4">
                      <div className="w-100 d-flex align-items-start">
                        <div className="text">
                          <p>
                            <span>Phone:</span>{" "}
                            <a href="tel://1234567920">(646)123-7654</a>
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                  <form
                    method="POST"
                    id="contactForm"
                    name="contactForm"
                    className="contactForm"
                  >
                    <div className="row">
                      <div className="col-md-12">
                        <div className="form-group">
                          <input
                            type="text"
                            className="form-control"
                            name="name"
                            id="name"
                            placeholder="Name"
                          />
                        </div>
                      </div>
                      <div className="col-md-12">
                        <div className="form-group">
                          <input
                            type="email"
                            className="form-control"
                            name="email"
                            id="email"
                            placeholder="Email"
                          />
                        </div>
                      </div>
                      <div className="col-md-12">
                        <div className="form-group">
                          <input
                            type="text"
                            className="form-control"
                            name="subject"
                            id="subject"
                            placeholder="Subject"
                          />
                        </div>
                      </div>
                      <div className="col-md-12">
                        <div className="form-group">
                          <textarea
                            name="message"
                            className="form-control mb-5 mt-2"
                            id="message"
                            cols="30"
                            rows="4"
                            placeholder="Create a message here"
                            style={{ height: "100px" }}
                          ></textarea>
                        </div>
                      </div>
                      <div className="col-md-12">
                        <div className="form-group text-center text-lg-start">
                          <a href="#0">Send Message</a>
                        </div>
                      </div>
                    </div>
                  </form>
                </div>
              </div>
              <div className="col-lg-6 d-none d-lg-flex align-items-stretch">
                <div
                  className="info-wrap w-100 p-5 img"
                  style={{
                    background: `url(${contact})`,
                    backgroundSize: "cover",
                  }}
                ></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default ContactForm;
