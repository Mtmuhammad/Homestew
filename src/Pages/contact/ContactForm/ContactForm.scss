@use "../../../Styles/abstracts/mixins" as *;
@use "../../../Styles/abstracts/variables" as *;

.contact {
  @include respond(sphone) {
    padding: 4rem 0px;
  }
}

.wrapper {
  width: 100%;
  overflow: hidden;
  border-radius: 10px;
  box-shadow: 0px 21px 41px -13px rgb(0 0 0 / 18%);
}

.no-gutters {
  margin-right: 0;
  margin-left: 0;
  padding-left: 0;
  padding-right: 0;
}

.contact-wrap {
  background-color: #fff;
  & h3 {
    font-size: 3rem;
  }

  & p {
    font-size: 1.7rem;
  }

  & .text {
    width: 100%;

    & span {
      color: rgba(0, 0, 0, 0.2);
      display: block;
      text-transform: uppercase;
      font-size: 13px;
      font-weight: 600;
    }

    & a {
      color: rgba(0, 0, 0, 0.7);
    }
  }
}

.contactForm {
  & .form-control {
    font-size: 1.6rem;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 0;
    height: 40px;
    background: transparent;
    color: rgba(0, 0, 0, 0.8) !important;
    border-radius: 0px;
    box-shadow: none !important;
  }

  & a {
    color: #fff;
    background-color: $primary-green;
    padding: 1.3rem;
    font-size: 1.5rem;
    text-transform: uppercase;
    border-radius: 5px;
    margin-top: 2rem;
    transition: all 0.3s;

    &:hover {
      color: $primary-font;
    }
  }
}
