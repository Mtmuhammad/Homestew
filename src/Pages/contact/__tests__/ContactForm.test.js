import React from "react";
import { render, screen } from "@testing-library/react";
import ContactForm from "../ContactForm/ContactForm";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", async () => {
    render(<ContactForm />);
  });

  test("should match snapshot", async () => {
    const { asFragment } = render(<ContactForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});

test("should render heading", () => {
  render(<ContactForm />);
  const heading = screen.getByRole("heading", { name: /contact us/i });
  expect(heading).toBeInTheDocument();
});
