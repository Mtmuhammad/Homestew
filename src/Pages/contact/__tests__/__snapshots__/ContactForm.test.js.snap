// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-md-12"
      >
        <div
          class="wrapper"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-lg-6"
            >
              <div
                class="contact-wrap w-100 p-md-5 p-4"
              >
                <h3
                  class="mb-3"
                >
                  Contact Us
                </h3>
                <p
                  class="mb-4"
                >
                  We're open for any suggestion or just to have a chat
                </p>
                <div
                  class="row mb-4"
                >
                  <div
                    class="col-md-4"
                  >
                    <div
                      class="w-100 d-flex align-items-start"
                    >
                      <div
                        class="text"
                      >
                        <p>
                          <span>
                            Address:
                          </span>
                           198 West 21th Street, Suite 721 New York NY 10016
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    class="col-md-4"
                  >
                    <div
                      class="w-100 d-flex align-items-start"
                    >
                      <div
                        class="text"
                      >
                        <p>
                          <span>
                            Email:
                          </span>
                           
                          <a
                            href="mailto:info@yoursite.com"
                          >
                            info@homestew.com
                          </a>
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    class="col-md-4"
                  >
                    <div
                      class="w-100 d-flex align-items-start"
                    >
                      <div
                        class="text"
                      >
                        <p>
                          <span>
                            Phone:
                          </span>
                           
                          <a
                            href="tel://1234567920"
                          >
                            (646)123-7654
                          </a>
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <form
                  class="contactForm"
                  id="contactForm"
                  method="POST"
                  name="contactForm"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-md-12"
                    >
                      <div
                        class="form-group"
                      >
                        <input
                          class="form-control"
                          id="name"
                          name="name"
                          placeholder="Name"
                          type="text"
                        />
                      </div>
                    </div>
                    <div
                      class="col-md-12"
                    >
                      <div
                        class="form-group"
                      >
                        <input
                          class="form-control"
                          id="email"
                          name="email"
                          placeholder="Email"
                          type="email"
                        />
                      </div>
                    </div>
                    <div
                      class="col-md-12"
                    >
                      <div
                        class="form-group"
                      >
                        <input
                          class="form-control"
                          id="subject"
                          name="subject"
                          placeholder="Subject"
                          type="text"
                        />
                      </div>
                    </div>
                    <div
                      class="col-md-12"
                    >
                      <div
                        class="form-group"
                      >
                        <textarea
                          class="form-control mb-5 mt-2"
                          cols="30"
                          id="message"
                          name="message"
                          placeholder="Create a message here"
                          rows="4"
                          style="height: 100px;"
                        />
                      </div>
                    </div>
                    <div
                      class="col-md-12"
                    >
                      <div
                        class="form-group text-center text-lg-start"
                      >
                        <a
                          href="#0"
                        >
                          Send Message
                        </a>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
            </div>
            <div
              class="col-lg-6 d-none d-lg-flex align-items-stretch"
            >
              <div
                class="info-wrap w-100 p-5 img"
                style="background: url(Contact.jpg); background-size: cover;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
